backend-rust/src/indexer.rs

```diff
@@ -2367,7 +2394,8 @@ impl PreparedBakerEvent {
                     delegator_id
                 )
                 .execute(tx.as_mut())
-                .await?;
+                .await?
+                .ensure_affected_rows_in_range(0..=1)?;
```

Comment:
Suggestion to add a comment at these `ensure_affected_rows_in_range` when to expect what number of row updates. E.g.
If `delegated_target_baker_id` is NULL (passive pool) we expect 0 row updates and otherwise 1.